[46기 장건웅]Add : Login UI 추가 및 인증 기능 구현

woong3e opened the pull request:
## 1. 본 PR이 우리 팀의 웹 서비스 제품성에 어떠한 기여를 하였고, 사용자에게 어떠한 기대효과를 전달하는지 작성해주세요.
- 내 PR이 제품 내 어떠한 기능적인 배경/전후맥락 가운데 개발되었나요?

Login UI를 추가하였습니다. 사용자가 로그인 할 수 있는 UI를 구성하였습니다. 

- 내 PR이 Merge 됨으로써 유저에게 전달되는 편익/기대효과는 무엇일까요?

사용자 권한으로 사이트를 이용할 수 있습니다.


<br />

## 2. 이 브랜치에서 어떤 내용을 개발했는지 큰 제목과 그리고 상세 내역을 적어주세요.

 - 로그인 UI 추가
 - 유효성 검사 추가, 이메일 input에 @ 포함, 비밀번호 input  길이 5이상
 - 로그인 인증 성공 시 main이동, localStorage에 token 값 저장, 실패시 alert창을 띄웁니다.
 - 계정만들기 버튼 클릭시 signup 페이지로 이동


<br />

## 3. 개발한 화면을 캡쳐해서 첨부 해 주세요. (drag & drop 또는 첨부파일 추가)
<br />

<img width="1512" alt="스크린샷 2023-05-31 오전 10 51 03" src="https://github.com/wecode-bootcamp-korea/46-1st-BestFriend-frontend/assets/86347667/71762b64-7438-43b2-a654-4c1c6c1622f2">

![스크린샷 2023-06-01 오후 4 50 24](https://github.com/wecode-bootcamp-korea/46-1st-BestFriend-frontend/assets/86347667/a3266188-0fd1-4515-91c3-aea4193da71c)

![스크린샷 2023-06-01 오후 4 51 11](https://github.com/wecode-bootcamp-korea/46-1st-BestFriend-frontend/assets/86347667/dfbe8379-6918-42ad-ba67-9168c61eca86)


<br />

## 4. 이 브랜치에서 개발하면서 느꼇던 개발 성장포인트를 적어주세요.

- 프론트,백엔드 간 로그인 흐름을 이해하였습니다
- jwt 및 accessToken를 이해하였습니다
- fetch 함수 구조를 이해하였습니다.
- useState의 렌더링과 함수형 업데이트를 이해하였습니다.
  <br />

woong3e committed:
login layout 완료

woong3e committed:
Modify: class명,input padding값 수정

sstaar91 commented on src/pages/Login/Login.js:
```diff
@@ -1,7 +1,20 @@
 const Login = () => {
-  return <div>Login</div>;
+  return (
+    <main className="login-container">
```
```suggestion
    <main className="login">
```

className은 컴포넌트의 이름과 동일하게 작성해주세요!

woong3e committed:
Feat: 로그인 UI 수정 및 로그인 인증 기능 구현

- 로그인 버튼 스타일을 수정했습니다.
- 이메일input에 @ 포함, 비밀번호input의 길이가 5이상 유효성 검사,실패시 alert창을 띄웁니다.
- 로그인 인증이 성공하면 main으로 이동 및 token을 local에 저장, 실패하면 alert창 띄웁니다.
- 계정만들기 버튼 클릭시 signup 페이지로 이동

woong3e committed:
Modify: 수정사항반영

sstaar91 commented on src/pages/Login/Login.js:
```diff
@@ -1,7 +1,93 @@
+
+  const handleUserInputs = e => {
+    const { name, value } = e.target;
+    setUserInputs({
+      ...userInputs,
+      [name]: value,
+    });
```
```suggestion
    setUserInputs(prev => ({
      ...prev,
      [name]: value,
    }));
```
이 부분 함수형 업데이트로 구현해주세요~

sstaar91 commented on src/pages/Login/Login.js:
```diff
@@ -1,7 +1,93 @@
+      return;
+    }
+
+    fetch('http://10.58.52.136:3000/users/signin', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        email: userId,
+        password: userPassword,
+      }),
+    })
```
이 부분 구두로 팁 전달 드리겠습니다!

sstaar91 reviewed (commented):
건웅님 수고하셨습니다!
수정 사항 반영해주세요~

woong3e committed:
Modify: 수정사항 반영

sstaar91 reviewed (approved):
LGTM 🚀🚀🚀🚀
